### ISSUE TITLE
hedera-mirror-monitor pod needs to be restarted after funding is reinstated

### ISSUE THREAD

**Issue by shezaan-hashgraph** [hashgraph/hedera-mirror-node#8873]
### Description

Deployment of the `hedera-mirror-monitor` service runs fine until it runs out of funds resulting in the need to re-instate the funds. After funds are re-instated the pod logs continue to spit out "INSUFFICIENT_PAYER_BALANCE" messages in pod logs until the pod is finally restarted. We have seen this happen multiple times.

### Steps to reproduce

1. Deploy the hedera-mirror-monitor service into k8s
2. Let it run for a few days until you start seeing "INSUFFICIENT_PAYER_BALANCE" in the pod logs.
3. Fund the account and note that the log messages don't go away until 

### Additional context

```bash
2024-06-20T18:35:49.453Z  INFO scheduling-3 c.h.m.m.p.PublishMetrics Scenario pinger published 49593285 transactions in 10d12h23m52s at 0.0/s. Errors: {ArrayIndexOutOfBoundsException=1, DEADLINE_EXCEEDED=578, DUPLICATE_TRANSACTION=1, INSUFFICIENT_PAYER_BALANCE=3207294, INTERNAL=2, IllegalArgumentException=21041550, IllegalStateException=705568, TimeoutException=1479049, UNAVAILABLE=179819} 
2024-06-20T18:35:49.453Z  INFO scheduling-3 c.h.m.m.p.PublishMetrics Scenario crypto published 50295 transactions in 10d12h23m52s at 0.0/s. Errors: {INSUFFICIENT_PAYER_BALANCE=3202, IllegalArgumentException=21055, TimeoutException=1477, UNAVAILABLE=190}
2024-06-20T18:35:59.454Z  INFO scheduling-4 c.h.m.m.p.PublishMetrics Scenario pinger published 49593285 transactions in 10d12h24m2s at 0.0/s. Errors: {ArrayIndexOutOfBoundsException=1, DEADLINE_EXCEEDED=578, DUPLICATE_TRANSACTION=1, INSUFFICIENT_PAYER_BALANCE=3207294, INTERNAL=2, IllegalArgumentException=21041550, IllegalStateException=706569, TimeoutException=1479049, UNAVAILABLE=179819} 
2024-06-20T18:35:59.454Z  INFO scheduling-4 c.h.m.m.p.PublishMetrics Scenario crypto published 50296 transactions in 10d12h24m2s at 0.0/s. Errors: {INSUFFICIENT_PAYER_BALANCE=3202, IllegalArgumentException=21055, TimeoutException=1477, UNAVAILABLE=190}
```

### Hedera network

other

### Version

v0.104.0

### Operating system

None

**Comment by steven-sheehy** [comment]
The PublishMetrics log is just a counter that only increases until the next restart. It will never go away without a restart and we won't change that. 

However, the monitor should recover after refilling the balance and cause the INSUFFICIENT_PAYER_BALANCE errors to stop increasing, but not decrease. You just mention the logs not going away, but is it not recovering as well?

**Comment by xin-hedera** [comment]
@shezaan-hashgraph verified monitor can recover by itself once the payer is funded to pay for new transactions.

```
2024-09-13T22:04:34.271Z  INFO scheduling-3 c.h.m.m.s.SubscribeMetrics GRPC scenario hcs received 61 responses in 8s at 0.0/s. Errors: {NOT_FOUND=3}
2024-09-13T22:04:34.271Z  INFO scheduling-3 c.h.m.m.s.SubscribeMetrics REST scenario transactionId received 1 responses in 8s at 0.0/s. Errors: {}
2024-09-13T22:04:44.276Z  INFO scheduling-1 c.h.m.m.p.PublishMetrics Scenario pinger published 153 transactions in 20s at 7.7/s. Errors: {INSUFFICIENT_PAYER_BALANCE=64}
2024-09-13T22:04:44.280Z  INFO scheduling-1 c.h.m.m.s.SubscribeMetrics GRPC scenario hcs received 138 responses in 18s at 6.9/s. Errors: {NOT_FOUND=3}
2024-09-13T22:04:44.280Z  INFO scheduling-1 c.h.m.m.s.SubscribeMetrics REST scenario transactionId received 3 responses in 18s at 0.2/s. Errors: {}
2024-09-13T22:04:54.284Z  INFO scheduling-3 c.h.m.m.s.SubscribeMetrics GRPC scenario hcs received 138 responses in 28s at 6.9/s. Errors: {NOT_FOUND=3}
2024-09-13T22:04:54.284Z  INFO scheduling-1 c.h.m.m.p.PublishMetrics Scenario pinger published 153 transactions in 30s at 7.7/s. Errors: {INSUFFICIENT_PAYER_BALANCE=247}
2024-09-13T22:04:54.285Z  INFO scheduling-3 c.h.m.m.s.SubscribeMetrics REST scenario transactionId received 3 responses in 28s at 0.2/s. Errors: {}
2024-09-13T22:05:04.291Z  INFO scheduling-2 c.h.m.m.s.SubscribeMetrics GRPC scenario hcs received 138 responses in 38s at 0.0/s. Errors: {NOT_FOUND=3}
2024-09-13T22:05:04.291Z  INFO scheduling-1 c.h.m.m.p.PublishMetrics Scenario pinger published 153 transactions in 40s at 0.0/s. Errors: {INSUFFICIENT_PAYER_BALANCE=542}
2024-09-13T22:05:04.291Z  INFO scheduling-2 c.h.m.m.s.SubscribeMetrics REST scenario transactionId received 3 responses in 38s at 0.0/s. Errors: {}
2024-09-13T22:05:14.298Z  INFO scheduling-3 c.h.m.m.s.SubscribeMetrics GRPC scenario hcs received 138 responses in 48s at 0.0/s. Errors: {NOT_FOUND=3}
2024-09-13T22:05:14.298Z  INFO scheduling-1 c.h.m.m.p.PublishMetrics Scenario pinger published 153 transactions in 50s at 0.0/s. Errors: {INSUFFICIENT_PAYER_BALANCE=842}
2024-09-13T22:05:14.298Z  INFO scheduling-3 c.h.m.m.s.SubscribeMetrics REST scenario transactionId received 3 responses in 48s at 0.0/s. Errors: {}
2024-09-13T22:05:24.305Z  INFO scheduling-1 c.h.m.m.s.SubscribeMetrics GRPC scenario hcs received 197 responses in 58s at 3.0/s. Errors: {NOT_FOUND=3}
2024-09-13T22:05:24.306Z  INFO scheduling-1 c.h.m.m.s.SubscribeMetrics REST scenario transactionId received 4 responses in 58s at 0.1/s. Errors: {}
2024-09-13T22:05:24.310Z  INFO scheduling-3 c.h.m.m.p.PublishMetrics Scenario pinger published 231 transactions in 1m at 3.9/s. Errors: {INSUFFICIENT_PAYER_BALANCE=1009}
2024-09-13T22:05:34.313Z  INFO scheduling-1 c.h.m.m.s.SubscribeMetrics GRPC scenario hcs received 385 responses in 1m8s at 3.0/s. Errors: {NOT_FOUND=3}
2024-09-13T22:05:34.314Z  INFO scheduling-1 c.h.m.m.s.SubscribeMetrics REST scenario transactionId received 8 responses in 1m8s at 0.1/s. Errors: {}
2024-09-13T22:05:34.313Z  INFO scheduling-4 c.h.m.m.p.PublishMetrics Scenario pinger published 429 transactions in 1m10s at 3.9/s. Errors: {INSUFFICIENT_PAYER_BALANCE=1166}
2024-09-13T22:05:44.320Z  INFO scheduling-1 c.h.m.m.s.SubscribeMetrics GRPC scenario hcs received 385 responses in 1m18s at 9.4/s. Errors: {NOT_FOUND=3}
2024-09-13T22:05:44.321Z  INFO scheduling-1 c.h.m.m.s.SubscribeMetrics REST scenario transactionId received 8 responses in 1m18s at 0.2/s. Errors: {}
2024-09-13T22:05:44.322Z  INFO scheduling-1 c.h.m.m.p.PublishMetrics Scenario pinger published 429 transactions in 1m20s at 9.9/s. Errors: {INSUFFICIENT_PAYER_BALANCE=1466}
2024-09-13T22:05:54.327Z  INFO scheduling-1 c.h.m.m.s.SubscribeMetrics GRPC scenario hcs received 385 responses in 1m28s at 9.4/s. Errors: {NOT_FOUND=3}
2024-09-13T22:05:54.327Z  INFO scheduling-3 c.h.m.m.p.PublishMetrics Scenario pinger published 429 transactions in 1m30s at 9.9/s. Errors: {INSUFFICIENT_PAYER_BALANCE=1767}
2024-09-13T22:05:54.328Z  INFO scheduling-1 c.h.m.m.s.SubscribeMetrics REST scenario transactionId received 8 responses in 1m28s at 0.2/s. Errors: {}
2024-09-13T22:06:04.333Z  INFO scheduling-4 c.h.m.m.s.SubscribeMetrics GRPC scenario hcs received 385 responses in 1m38s at 0.0/s. Errors: {NOT_FOUND=3}
2024-09-13T22:06:04.334Z  INFO scheduling-4 c.h.m.m.s.SubscribeMetrics REST scenario transactionId received 8 responses in 1m38s at 0.0/s. Errors: {}
2024-09-13T22:06:04.333Z  INFO scheduling-3 c.h.m.m.p.PublishMetrics Scenario pinger published 439 transactions in 1m40s at 0.5/s. Errors: {INSUFFICIENT_PAYER_BALANCE=1999}
2024-09-13T22:06:14.336Z  INFO scheduling-2 c.h.m.m.p.PublishMetrics Scenario pinger published 738 transactions in 1m50s at 0.5/s. Errors: {INSUFFICIENT_PAYER_BALANCE=1999}
2024-09-13T22:06:14.336Z  INFO scheduling-4 c.h.m.m.s.SubscribeMetrics GRPC scenario hcs received 675 responses in 1m48s at 0.0/s. Errors: {NOT_FOUND=3}
2024-09-13T22:06:14.337Z  INFO scheduling-4 c.h.m.m.s.SubscribeMetrics REST scenario transactionId received 14 responses in 1m48s at 0.0/s. Errors: {}
2024-09-13T22:06:24.344Z  INFO scheduling-2 c.h.m.m.p.PublishMetrics Scenario pinger published 1041 transactions in 2m at 30.1/s. Errors: {INSUFFICIENT_PAYER_BALANCE=1999}
2024-09-13T22:06:24.344Z  INFO scheduling-1 c.h.m.m.s.SubscribeMetrics GRPC scenario hcs received 979 responses in 1m58s at 29.7/s. Errors: {NOT_FOUND=3}
2024-09-13T22:06:24.345Z  INFO scheduling-1 c.h.m.m.s.SubscribeMetrics REST scenario transactionId received 18 responses in 1m58s at 0.5/s. Errors: {}
2024-09-13T22:06:34.350Z  INFO scheduling-2 c.h.m.m.p.PublishMetrics Scenario pinger published 1341 transactions in 2m10s at 30.1/s. Errors: {INSUFFICIENT_PAYER_BALANCE=1999}
2024-09-13T22:06:34.350Z  INFO scheduling-4 c.h.m.m.s.SubscribeMetrics GRPC scenario hcs received 1278 responses in 2m8s at 29.7/s. Errors: {NOT_FOUND=3}
2024-09-13T22:06:34.350Z  INFO scheduling-4 c.h.m.m.s.SubscribeMetrics REST scenario transactionId received 27 responses in 2m8s at 0.5/s. Errors: {}
2024-09-13T22:06:44.355Z  INFO scheduling-2 c.h.m.m.p.PublishMetrics Scenario pinger published 1642 transactions in 2m20s at 30.1/s. Errors: {INSUFFICIENT_PAYER_BALANCE=1999}
2024-09-13T22:06:44.355Z  INFO scheduling-3 c.h.m.m.s.SubscribeMetrics GRPC scenario hcs received 1576 responses in 2m18s at 29.9/s. Errors: {NOT_FOUND=3}
2024-09-13T22:06:44.355Z  INFO scheduling-3 c.h.m.m.s.SubscribeMetrics REST scenario transactionId received 33 responses in 2m18s at 0.8/s. Errors: {}
2024-09-13T22:06:54.362Z  INFO scheduling-2 c.h.m.m.p.PublishMetrics Scenario pinger published 1941 transactions in 2m30s at 30.1/s. Errors: {INSUFFICIENT_PAYER_BALANCE=1999}
2024-09-13T22:06:54.362Z  INFO scheduling-4 c.h.m.m.s.SubscribeMetrics GRPC scenario hcs received 1877 responses in 2m28s at 29.9/s. Errors: {NOT_FOUND=3}
2024-09-13T22:06:54.363Z  INFO scheduling-4 c.h.m.m.s.SubscribeMetrics REST scenario transactionId received 43 responses in 2m28s at 0.8/s. Errors: {}
 2024-09-13T22:07:04.365Z  INFO scheduling-2 c.h.m.m.p.PublishMetrics Scenario pinger published 2242 transactions in 2m40s at 30.0/s. Errors: {INSUFFICIENT_PAYER_BALANCE=1999}
2024-09-13T22:07:04.365Z  INFO scheduling-3 c.h.m.m.s.SubscribeMetrics GRPC scenario hcs received 2177 responses in 2m38s at 30.1/s. Errors: {NOT_FOUND=3}
2024-09-13T22:07:04.366Z  INFO scheduling-3 c.h.m.m.s.SubscribeMetrics REST scenario transactionId received 53 responses in 2m38s at 1.0/s. Errors: {}
 2024-09-13T22:07:14.372Z  INFO scheduling-2 c.h.m.m.p.PublishMetrics Scenario pinger published 2540 transactions in 2m50s at 30.0/s. Errors: {INSUFFICIENT_PAYER_BALANCE=1999}
2024-09-13T22:07:14.373Z  INFO scheduling-2 c.h.m.m.s.SubscribeMetrics GRPC scenario hcs received 2476 responses in 2m48s at 30.1/s. Errors: {NOT_FOUND=3}
2024-09-13T22:07:14.373Z  INFO scheduling-2 c.h.m.m.s.SubscribeMetrics REST scenario transactionId received 58 responses in 2m48s at 1.0/s. Errors: {}
```

This can be observed by the log rows that the number of `INSUFFICIENT_PAYER_BALANCE` error stopped increasing at the end, and the number of published messages, the number os received messages by both GRPC and REST started to increase.
